R/derive_param_clinbenefit.R

```diff
@@ -0,0 +1,256 @@
+#' Derive a Clinical Benefit Parameter
+#'
+#' Add a clinical benefit/disease control parameter to the input dataset.
+#'
+#' @details
+#' Clinical benefit/disease control is first identified for looking for subjects
+#' having objective response, and then derived for subjects that have at least
+#' one evaluable non-PD response assessment after a specified amount of time
+#' from a reference date.
+#'
+#' @param dataset Input dataset. This is the dataset to which the clinical
+#' benefit rate parameter will be added.
+#'
+#'   The variables `PARAMCD`, `AVALC`, `ADT`, and those specified by the `by_vars`
+#'   parameter are expected.
+#'
+#' @param dataset_adsl ADSL dataset used as input for populating `subject_keys`
+#' in input dataset.
+#'
+#'   The variables specified by the `subject_keys` parameter and the `reference_date`
+#'   parameter are expected.
+#'
+#' @param filter_source Filter condition in `dataset` that represents records
+#' for overall disease response assessment for a subject at a given timepoint,
+#' e.g. `PARAMCD == "OVR"` or `PARAMCD == "OVRLRESP"`.
+#'
+#' @param source_resp A `date_source` object specifying the dataset, date variable,
+#' and filter condition used to identify objective response status.
+#'
+#' @param source_pd A `date_source` object specifying the dataset, date variable,
+#' and filter condition used to identify disease progression.
+#'
+#' @param source_datasets  A named list of data sets is expected.
+#'
+#'  The list must contain the names provided by the `dataset_name` field of the
+#'  `admiral::date_source()` objects specified for `source_pd` and `source_resp`.
+#'
+#' @param by_vars A named list returned by `vars()` used together with `subject_keys`
+#' to identify groupings within which the earliest date of clinical benefit rate
+#' status will be chosen.
+#'
+#' @param reference_date Name of variable representing the index date for
+#' `ref_start_window`. A variable providing a date. An unquoted symbol is expected.
+#'
+#' @param ref_start_window Integer representing number of days from `reference_date`
+#' that must elapse before an evaluable non-PD assessment counts toward determining
+#' clinical benefit.
+#'
+#' @param set_values_to A named list returned by `vars()` containing new variables
+#' and their static value to be populated for the clinical benefit rate parameter
+#' records, e.g. `vars(PARAMCD = "CBR", PARAM = "Clinical Benefit Rate")`.
+#'
+#' @param subject_keys A named list returned by `vars()` containing variables
+#' used to uniquely identify subjects.
+#'
```

Comment:
The `@return` and the `@keywords` tag are missing.

Reply:
updated